sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -37,7 +37,8 @@ case class BatchScanExec(
     @transient scan: Scan,
     runtimeFilters: Seq[Expression],
     keyGroupedPartitioning: Option[Seq[Expression]] = None,
-    ordering: Option[Seq[SortOrder]] = None) extends DataSourceV2ScanExecBase {
+    ordering: Option[Seq[SortOrder]] = None,
+    table: Option[String] = None) extends DataSourceV2ScanExecBase {
```

Comment:
since we are adding a new parameter, shall we add `Table` directly? in case we may need to access other methods of `Table` in the future.

Reply:
BTW, since `Scan` can only be created from `Table`, when this new parameter can be `None`?

Reply:
> since we are adding a new parameter, shall we add `Table` directly? in case we may need to access other methods of `Table` in the future.

I initially had added `Table` https://github.com/apache/spark/pull/37505/commits/710944e86c2254708a207855bd65d1ba09463ec5
But then there were test failures with `NotSerializableException` exceptions. 
I made `CaseInsensitiveStringMap` serializable and ran the failed test locally but the test failure persisted. 
So a quick-fix was to simply pass the table name as String instead of `Table`.




```
sbt:spark-avro> testOnly org.apache.spark.sql.avro.AvroV2LogicalTypeSuite -- -z "Logical type: write Decimal with BYTES type"
.
.
[info] AvroV2LogicalTypeSuite:
[info] - Logical type: write Decimal with BYTES type *** FAILED *** (4 seconds, 704 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task not serializable: java.io.NotSerializableException: org.apache.spark.sql.util.CaseInsensitiveStringMap
[info]   at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2706)
[info]   at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2642)
[info]   at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2641)
```

Reply:
> BTW, since `Scan` can only be created from `Table`, when this new parameter can be `None`?

It is `None` while instantiating `BatchScanExec` in the following two test cases in `SQLAppStatusListenerSuite`
- "SPARK-34338: Report metrics from Datasource v2 scan" https://github.com/apache/spark/blob/1103343e71fbcb478fa41941c87d2c28b0c09281/sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListenerSuite.scala#L843
- "SPARK-39635: Report driver metrics from Datasource v2 scan" https://github.com/apache/spark/blob/1103343e71fbcb478fa41941c87d2c28b0c09281/sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListenerSuite.scala#L881

Reply:
After making `CaseInsensitiveStringMap` serializable the test fails with `InvalidClassException` for `AvroTable` during deserialization

```java
[info] AvroV2LogicalTypeSuite:
[info] - Logical type: write Decimal with BYTES type *** FAILED *** (4 seconds, 852 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1.0 (TID 1) (192.168.0.109 executor driver): java.io.InvalidClassException: org.apache.spark.sql.v2.avro.AvroTable; no valid constructor
[info]  at java.base/java.io.ObjectStreamClass$ExceptionInfo.newInvalidClassException(ObjectStreamClass.java:159)
```

Reply:
to fix the serialization issue, shall we just mark it as `@transient`?

Reply:
Marking it transient solves the issue - thank you for the suggestion :-)